src/Instant.php

```diff
@@ -77,7 +77,14 @@ public static function of(int $epochSecond, int $nanoAdjustment = 0): Instant
 
     public static function epoch(): Instant
     {
-        return new Instant(0, 0);
+        /** @var Instant|null $epoch */
+        static $epoch;
+
+        if ($epoch) {
+            return $epoch;
+        }
+
+        return $epoch = new Instant(0, 0);
```

Comment:
```php
        /** @var Instant|null $epoch */
        static $epoch;

        if (!isset($epoch)) {
            $epoch = new Instant(0, 0);
        }

        return $epoch;
```
I think this variant is simpler, more cache-way

Reply:
Agree. That's how I spontaneously wrote the PR, then I changed it to match the current code style..

Reply:
I don't mind if you want to change this.
If you do, please use `if ($epoch === null)` instead of `isset()`, which wouldn't alert on undefined variable!